### ISSUE TITLE
Disable resize border blocking interactions when maximized

### ISSUE THREAD

**Issue by ArinFaraj** [bitsdojo/bitsdojo_window#14]
Currently for resizing we have a border inside the app window, which is fine when the app is not maximized,
but when it is maximized, it blocks click interactions for close buttons and so on around the app for the width of the border.
![image](https://user-images.githubusercontent.com/13029655/103389334-4a723480-4b1f-11eb-8666-632ed0ff67d8.png)
and because the border has no other usages when maximized I suggest disabling it entirely until the app size is restored or no longer maximized

**Comment by ArinFaraj** [comment]
its caused by the margin that is being added in `handle_nccalcsize` inside `bitsdojo_window.cpp`
``` cpp
if (isMaximized)
    {
        int resizeMargin = getResizeMargin(window);
        params->rgrc[0].left += resizeMargin;
        params->rgrc[0].top += resizeMargin;
        params->rgrc[0].right -= resizeMargin;
        params->rgrc[0].bottom -= resizeMargin;
    }
```
if possible instead we could try giving the margin to the child window but I'm so good in c++ 
if anyone with more knowledge can try this out please do and let us know the results

**Comment by bitsdojo** [comment]
Fixed by #15
